google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -143,118 +180,91 @@ def create(
                 be immediately returned and synced when the Future has completed.
+
         Returns:
             dataset (Dataset):
                 Instantiated representation of the managed dataset resource.
+
+        Raises:
+            ValueError if import_schema_uri is set when creating a tabular dataset.
         """
-        utils.validate_display_name(display_name)
 
-        is_tabular_dataset_metadata = (
-            metadata_schema_uri == schema.dataset.metadata.tabular
-        )
+        utils.validate_display_name(display_name)
 
-        if bool(bq_source) and not is_tabular_dataset_metadata:
-            raise ValueError(
-                "A tabular metadata_schema uri must be used when using a BigQuery source"
-            )
+        # Validate metadata_schema_uri and import_schema_uri for Dataset class
+        cls._validate_import_schema_uri(import_schema_uri)
 
-        api_client = cls._instantiate_client(location=location, credentials=credentials)
+        if metadata_schema_uri == schema.dataset.metadata.tabular:
+            datasource = datasources.TabularDatasource(gcs_source, bq_source)
+        else:
+            datasource = datasources.NonTabularDatasource()
+            if import_schema_uri:
+                datasource = datasources.NonTabularDatasourceImportable(
+                    gcs_source, import_schema_uri, data_item_labels
+                )
 
-        # If this is tabular enrich the dataset metadata with source
-        dataset_metadata = {}
-        if is_tabular_dataset_metadata:
-            if gcs_source:
-                dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
-            elif bq_source:
-                dataset_metadata = {
-                    "input_config": {"bigquery_source": {"uri": bq_source}}
-                }
-
-        return cls._create_and_import(
-            api_client=api_client,
+        return cls._create_encapsulated(
             display_name=display_name,
-            parent=initializer.global_config.common_location_path(
-                project=project, location=location
-            ),
             metadata_schema_uri=metadata_schema_uri,
-            dataset_metadata=dataset_metadata,
-            request_metadata=metadata,
+            datasource=datasource,
             labels=labels,
-            project=project or initializer.global_config.project,
-            location=location or initializer.global_config.location,
-            credentials=credentials or initializer.global_config.credentials,
-            gcs_source=gcs_source,
-            import_schema_uri=import_schema_uri,
-            data_items_labels=data_items_labels,
+            project=project,
+            location=location,
+            credentials=credentials,
+            request_metadata=request_metadata,
             sync=sync,
         )
 
     @classmethod
     @base.optional_sync()
-    def _create_and_import(
+    def _create_encapsulated(
```

Comment:
I thought we could just keep the `_create_and_import` name since you are replacing it anyways. No strong opinion from my side, so feel free to decide.